java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -53,19 +60,83 @@ protected MethodMatchers getMethodInvocationMatchers() {
   }
 
   @Override
   protected void onConstructorFound(NewClassTree newClassTree) {
+    if (newClassTree.classBody() != null) {
+      return;
+    }
     if(isBigIntegerPotentiallyBiggerThanLong(newClassTree)) {
       return;
     }
-    reportIssue(newClassTree.identifier(), "Remove this \"" + newClassTree.symbolType().name() + "\" constructor");
+    ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) context).newIssue())
+      .forRule(this)
+      .onTree(newClassTree.identifier())
+      .withMessage("Remove this \"" + newClassTree.symbolType().name() + "\" constructor")
+      .withQuickFix(() -> createQuickFix(newClassTree))
+      .report();
+  }
+
+  private static JavaQuickFix createQuickFix(NewClassTree newClassTree) {
+    if (newClassTree.symbolType().is(STRING)) {
+      return createStringQuickFix(newClassTree);
+    } else if (newClassTree.symbolType().is(BIG_INTEGER)) {
+      return createBigIntegerQuickFix(newClassTree);
+    }
+    return createDefaultQuickFix(newClassTree);
+  }
+
+  private static JavaQuickFix createDefaultQuickFix(NewClassTree newClassTree) {
+    String typeName = newClassTree.symbolType().name();
+    return JavaQuickFix.newQuickFix("Replace with " + typeName + ".valueOf")
+        .addTextEdit(JavaTextEdit.replaceTextSpan(
+          AnalyzerMessage.textSpanBetween(
+            newClassTree.firstToken(), true,
+            newClassTree.arguments().openParenToken(), true),
+          typeName + ".valueOf("))
+        .build();
+  }
+
+  private static JavaQuickFix createBigIntegerQuickFix(NewClassTree newClassTree) {
+    AnalyzerMessage.TextSpan constructor = AnalyzerMessage.textSpanFor(newClassTree);
+    AnalyzerMessage.TextSpan argument = AnalyzerMessage.textSpanFor(newClassTree.arguments().get(0));
+    return JavaQuickFix.newQuickFix("Replace with BigInteger.valueOf")
+        .addTextEdit(JavaTextEdit.replaceTextSpan(new AnalyzerMessage.TextSpan(
+          constructor.startLine, constructor.startCharacter,
+          argument.startLine, argument.startCharacter + 1 /* remove the first " */),
+          "BigInteger.valueOf("))
+        .addTextEdit(JavaTextEdit.replaceTextSpan(new AnalyzerMessage.TextSpan(
+          argument.endLine, argument.endCharacter - 1 /* remove the last " */,
+          constructor.endLine, constructor.endCharacter),
+          "L)"))
+        .build();
+  }
+
+  private static JavaQuickFix createStringQuickFix(NewClassTree newClassTree) {
+    if (newClassTree.arguments().isEmpty()) {
+      return JavaQuickFix.newQuickFix("Replace with \"\"")
+          .addTextEdit(JavaTextEdit.replaceTree(newClassTree, "\"\""))
+          .build();
+    } else if (newClassTree.arguments().get(0).is(Tree.Kind.STRING_LITERAL)) {
+      return JavaQuickFix.newQuickFix("Remove \"new String\"")
+          .addTextEdit(JavaTextEdit.replaceTextSpan(AnalyzerMessage.textSpanBetween(
```

Comment:
since both tokens are included, you can directly call `replaceBetweenTree`:

```java
.addTextEdit(JavaTextEdit.replaceBetweenTree(newClassTree.firstToken(), arguments().openParenToken(), ""))
.build();
```